src/equations/serre_green_naghdi_1d.jl

```diff
@@ -0,0 +1,960 @@
+        end
+        @. M_h_p_h_bx2 = h + factor * h * b_x^2
+        scale_by_mass_matrix!(M_h_p_h_bx2, D)
+        inv3 = 1 / 3
+        @. M_h3_3 = inv3 * h^3
+        scale_by_mass_matrix!(M_h3_3, D)
+        @. M_h2_bx = 0.5 * h^2 * b_x
+        scale_by_mass_matrix!(M_h2_bx, D)
+        system_matrix = Symmetric(Diagonal(M_h_p_h_bx2)
+                                  +
+                                  Dmat_minus' * (Diagonal(M_h3_3) * Dmat_minus
+                                   -
+                                   Diagonal(M_h2_bx))
+                                  -
+                                  Diagonal(M_h2_bx) * Dmat_minus)
+
+        factorization = cholesky(system_matrix)
+
+        cache = (; h, h_x, v_x, v_x_upwind, h_hpb_x, b, b_x, hv_x, v2_x,
+                 h2_v_vx_x, h_vx_x, p_h, p_0, p_x, tmp,
+                 M_h_p_h_bx2, M_h3_3, M_h2_bx,
+                 D, Dmat_minus, factorization)
+    else
+        if D isa FourierDerivativeOperator
+            Dmat = Matrix(D)
+
+            cache = (; h, h_x, v_x, h_hpb_x, b, b_x, hv_x, v2_x,
+                     h2_v_vx_x, h_vx_x, p_h, p_x, tmp,
+                     M_h_p_h_bx2, M_h3_3, M_h2_bx,
+                     D, Dmat)
+        else
+            Dmat = sparse(D)
+
+            # Floating point errors accumulate a bit and the system matrix
+            # is not necessarily perfectly symmetric but only up to
+            # round-off errors. We wrap it here to avoid issues with the
+            # factorization.
+            if equations.bathymetry isa BathymetryMildSlope
+                factor = 0.75
+            elseif equations.bathymetry isa BathymetryVariable
+                factor = 1.0
+            else
+                throw(ArgumentError("Unsupported bathymetry type $(equations.bathymetry)"))
+            end
+            @. M_h_p_h_bx2 = h + factor * h * b_x^2
+            scale_by_mass_matrix!(M_h_p_h_bx2, D)
+            inv3 = 1 / 3
+            @. M_h3_3 = inv3 * h^3
+            scale_by_mass_matrix!(M_h3_3, D)
+            @. M_h2_bx = 0.5 * h^2 * b_x
+            scale_by_mass_matrix!(M_h2_bx, D)
+            system_matrix = Symmetric(Diagonal(M_h_p_h_bx2)
+                                      +
+                                      Dmat' * (Diagonal(M_h3_3) * Dmat
+                                               -
+                                               Diagonal(M_h2_bx))
+                                      -
+                                      Diagonal(M_h2_bx) * Dmat)
+
+            factorization = cholesky(system_matrix)
+
+            cache = (; h, h_x, v_x, h_hpb_x, b, b_x, hv_x, v2_x,
+                     h2_v_vx_x, h_vx_x, p_h, p_x, tmp,
+                     M_h_p_h_bx2, M_h3_3, M_h2_bx,
+                     D, Dmat, factorization)
+        end
+    end
+
+    if equations.bathymetry isa BathymetryVariable
+        ψ = zero(h)
+        cache = (; cache..., ψ)
```

Comment:
```suggestion
        psi = zero(h)
        cache = (; cache..., psi)
```
I usually follow the style guide of Trixi.jl to not include any non-ASCII characters.